src/app/gauge/gauge.component.ts

```diff
@@ -64,12 +64,19 @@ export class GaugeComponent implements OnInit, AfterViewInit, GaugeProps {
   }
 
   ngOnInit(): void {
+    if (!this.start) {
+      this.start = Config.DEF_START;
+    }
+    if (!this.end) {
+      this.end = Config.DEF_END;
```

Comment:
`start` and `end` are still using the `Config` object not the property, so you won't be able to change them even if you pass them to the config `@Input`.

Reply:
if I remove these line, user should define start and end with [start] and [end] inputs, there is no default value If I remove these lines.

Reply:
Indeed. I meant that they are still using `Config.DEF_START` and `Config.DEF_END` (i.e. the default configuration). Lines should be still there but they must use `this.config` instead.